rfcs/0005-moxy-carousel.md

```diff
@@ -0,0 +1,195 @@
+- Start Date: 2020-01-21
+- Project: [react-carousel](https://github.com/moxystudio/react-carousel)
+- Reference Issues: [OFICINA-43](https://moxy.atlassian.net/browse/OFICINA-43)
+- Implementation PR: <https://gitlab.com/moxystudio/inoveretail/inovretail-www/merge_requests/102>
+
+# Summary
+
+This proposal is to use [react-multi-carousel](https://github.com/YIZHUANG/react-multi-carousel) as a baseline to develop our own `Carousel` component. Comparing to other npm packages, namely [nuka-carousel](https://github.com/FormidableLabs/nuka-carousel), [react-slick](https://github.com/akiran/react-slick) and [material-auto-rotating-carousel](https://github.com/TeamWertarbyte/material-auto-rotating-carousel), it's the one that better fits our needs, which are:
+
+- Infinite mode
+- Center mode
+- Control mode (arrows/dots)
+- Custom animation
+- AutoPlay mode
+- Responsive
+- Swipe to slide
+- Mouse drag to slide
+- Keyboard control to slide
+- Custom styling
+- Accessibility support
+- Framework agnostic - *this is the only one not offered by `react-multi-carrousel`*
+
+# Basic example
+
+This example was made using [next-with-moxy](https://github.com/moxystudio/next-with-moxy/) boilerplate.
+
+Preview:
+
+![react-carousel](https://user-images.githubusercontent.com/8797405/72811982-9dc5dc80-3c58-11ea-9a26-aaed7a2e25c4.gif)
+
+- `Home.js`
+
+```jsx
+import React from 'react';
+import Carousel from '@moxy/react-carousel';
+import classNames from 'classnames';
+
+import styles from './Home.module.css';
+
+const responsive = {
+    desktop: {
+        breakpoint: { max: 3000, min: 1024 },
+        items: 3,
+        slidesToSlide: 1, // Optional, default to 1.
+        partialVisibilityGutter: 40,
+    },
+    tablet: {
+        breakpoint: { max: 1024, min: 464 },
+        items: 2,
+        slidesToSlide: 2, // Optional, default to 1.
+    },
+    mobile: {
+        breakpoint: { max: 464, min: 0 },
+        items: 1,
+        slidesToSlide: 1, // Optional, default to 1.
+    },
+};
+
+const Slide = ({ className, title }) => (
+    <div className={ classNames(styles.slide, className) }>{title}</div>
+);
+
+const slidesData = [
+    { title: 'item 1', className: styles.firstSlide },
+    { title: 'item 2', className: styles.secondSlide },
+    { title: 'item 3', className: styles.thirdSlide },
+    { title: 'item 4', className: styles.forthSlide },
+];
+
+const Home = () => (
+    <Carousel
+        swipeable={ false }
+        draggable={ false }
+        showDots
+        responsive={ responsive }
+        ssr // Means to render carousel on server-side.
+        infinite
+        autoplay={ false }
+        autoPlaySpeed={ 1000 }
```

Comment:
It doesn't make much sense to set `autoPlaySpeed` when `autoplay` is `false`...
Should we get rid of `autoplay` and use just keep `autoPlaySpeed` around instead? If `autoPlaySpeed` is `undefined`, the carousel would not autoplay. I think we should also rename the prop to `autoPlaySpeedMs` or similar to indicate the time is provided in milliseconds.

Reply:
Agreed! I will take a look into the source code to check how feasible this is